plugins/builtin/target/azure-vmss/plugin/plugin.go

```diff
@@ -203,28 +251,99 @@ func (t *TargetPlugin) calculateDirection(vmssDesired, strategyDesired int64) (i
 
 // processInstanceView updates the status object based on the details within
 // the vmss instances.
-func processInstanceView(instanceView compute.VirtualMachineScaleSetInstanceView, status *sdk.TargetStatus) {
+func processInstanceView(instanceView *armcompute.VirtualMachineScaleSetsClientGetInstanceViewResponse, status *sdk.TargetStatus) {
 
-	for _, instanceStatus := range *instanceView.VirtualMachine.StatusesSummary {
-		if *instanceStatus.Code != "ProvisioningState/succeeded" {
-			status.Ready = false
-		}
+	if instanceView == nil || len(instanceView.VirtualMachineScaleSetInstanceView.Statuses) == 0 {
+		return
 	}
 
 	latestTime := int64(math.MinInt64)
-	for _, instanceStatus := range *instanceView.Statuses {
-		if *instanceStatus.Code != "ProvisioningState/succeeded" {
-			status.Ready = false
-		}
+	if instanceView.VirtualMachineScaleSetInstanceView.Statuses != nil {
+		for _, instanceStatus := range instanceView.VirtualMachineScaleSetInstanceView.Statuses {
+			if instanceStatus.Code != nil && *instanceStatus.Code != "ProvisioningState/succeeded" {
+				status.Ready = false
+			}
 
-		// Time isn't always populated, especially if the activity has not yet
-		// finished :).
-		if instanceStatus.Time != nil {
-			currentTime := instanceStatus.Time.Time.UnixNano()
-			if currentTime > latestTime {
-				latestTime = currentTime
-				status.Meta[sdk.TargetStatusMetaKeyLastEvent] = strconv.FormatInt(currentTime, 10)
+			// Time isn't always populated, especially if the activity has not yet
+			// finished :).
+			if instanceStatus.Time != nil {
+				currentTime := instanceStatus.Time.UnixNano()
+				if currentTime > latestTime {
+					latestTime = currentTime
+					status.Meta[sdk.TargetStatusMetaKeyLastEvent] = strconv.FormatInt(currentTime, 10)
+				}
 			}
 		}
 	}
 }
+
+func (t *TargetPlugin) processInstanceViewFlexible(vms []string, resourceGroup string, status *sdk.TargetStatus) {
+	readyInstances := []string{}
+	var mu sync.Mutex
+
+	t.logger.Debug("Processing VM instance views for Flexible VMSS", "vm_count", len(vms))
+	t.readyFlexibleInstances = []string{}
+
+	if len(vms) == 0 {
+		t.logger.Debug("No VMs found in the VMSS, skipping instance view processing.")
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var wg sync.WaitGroup
+	requests := make(chan struct{}, concurrentInstanceViewRequestLimit)
+	var notReady atomic.Bool
+
+	for _, vmName := range vms {
+		if ctx.Err() != nil {
+			t.logger.Debug("Context cancelled, stopping further processing of VMs")
+			break
+		}
+
+		wg.Add(1)
+		requests <- struct{}{}
+
+		go func(vm string) {
+			defer wg.Done()
+			defer func() { <-requests }()
+
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			instanceView, err := t.vm.InstanceView(ctx, resourceGroup, vm, nil)
+			if err != nil {
+				if strings.Contains(err.Error(), "RESPONSE 404") {
+					t.logger.Debug("VM not found", "vm_name", vm)
```

Comment:
Should this be treated as "not ready" too? What about a non-404/non-context error?